packages/nvidia_nat_mcp/src/nat/plugins/mcp/client_base.py

```diff
@@ -281,6 +314,47 @@ async def _with_reconnect(self, coro):
                 return await coro()
             raise
 
+    async def _has_cached_auth_token(self) -> bool:
+        """
+        Check if we have a cached, non-expired authentication token.
+
+        Returns:
+            bool: True if we have a valid cached token, False if authentication may be needed
+        """
+        if not self._auth_provider:
+            return True  # No auth needed
+
+        try:
+            # Check if OAuth2 provider has tokens cached
```

Comment:
ooh this is a bit tricky...We need the auth_provider to provide us with an API for it. But I am OK with it for now :)